src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/bfaction/BfActionReportTest.java

```diff
@@ -0,0 +1,265 @@
+        submitEvent.setState(ACCEPTED_STATE);
+        submitEvent.setCaseData(caseData);
+        submitEvents.add(submitEvent);
+
+        var resultListingData = bfActionReport.runReport(listingDetails, submitEvents);
+        var expectedBfDateCount = 2;
+        assertEquals(expectedBfDateCount, resultListingData.getBfDateCollection().size());
+    }
+
+    @Test
+    public void shouldReturnBfActionsSortedByBfDate() {
+        listingData.setListingDateFrom("2019-12-08");
+        listingData.setListingDateTo("2019-12-25");
+        listingData.setHearingDateType(RANGE_HEARING_DATE_TYPE);
+        listingDetails.setCaseData(listingData);
+        List<BFActionTypeItem> items = getBFActionTypeItemsWithClearedBfAction();
+
+        var bfActionTypeItem6 = new BFActionTypeItem();
+        bfActionTypeItem6.setId("456");
+        var bFActionType6 = new BFActionType();
+        bFActionType6.setCwActions("Interlocutory new order requested");
+        bFActionType6.setBfDate("2019-12-08");
+        bFActionType6.setDateEntered("2019-11-20");
+        bFActionType6.setNotes("test non-cleared bf sixth");
+        bfActionTypeItem6.setValue(bFActionType6);
+        items.add(bfActionTypeItem6);
+        caseData.setBfActions(items);
+        var bfActionReport = new BfActionReport();
+
+        submitEvent.setCaseId(2);
+        submitEvent.setState(ACCEPTED_STATE);
+        submitEvent.setCaseData(caseData);
+        submitEvents.add(submitEvent);
+
+        var resultListingData = bfActionReport.runReport(listingDetails, submitEvents);
+        // bFActionType3 is added last. But it has the earliest bfDate. As the returned listingData from
+        // bfActionReport.runReport method call should be ordered by bfDate, bFActionType3
+        // should be the first element
+        var bFActionType3 = items.get(2).getValue();
+        var firstBFDateTypeItem = resultListingData.getBfDateCollection().get(0).getValue();
+        assertEquals(bFActionType3.getBfDate(), firstBFDateTypeItem.getBroughtForwardDate());
+        assertEquals(bFActionType3.getCwActions(), firstBFDateTypeItem.getBroughtForwardAction());
+        assertEquals(bFActionType3.getNotes(), firstBFDateTypeItem.getBroughtForwardDateReason());
+        assertEquals(bFActionType3.getDateEntered(), firstBFDateTypeItem.getBroughtForwardEnteredDate());
+        assertEquals(bFActionType3.getCleared(), firstBFDateTypeItem.getBroughtForwardDateCleared());
+    }
+
+    private List<BFActionTypeItem> getBFActionTypeItemsWithClearedBfAction() {
+
+        var bfActionTypeItem3 = new BFActionTypeItem();
+        bfActionTypeItem3.setId("116");
+        var bFActionType3 = new BFActionType();
+        bFActionType3.setCwActions("Interlocutory order requested");
+        bFActionType3.setBfDate("2019-12-13");
+        bFActionType3.setDateEntered("2019-11-20");
+        bFActionType3.setNotes("test non-cleared bf two");
+        bfActionTypeItem3.setValue(bFActionType3);
```

Comment:
Are these BF's supposed to be cleared BF's? The 'cleared' field is not being populated so these BF's are not cleared

Reply:
The ones returned do not have to be cleared.  Infact, cleared BF entries should not be returned.

Reply:
I'm confused because the method name suggests that the BFs are cleared

Reply:
I see. The method naming is bad. I sort it.